src/lexer.rs

```diff
@@ -351,12 +363,23 @@ impl<'a> Lexer<'a> {
         self.text_token(start, TokenKind::String)
     }
 
-    fn read_integer(&mut self) -> Option<Token<'a>> {
+    fn read_number(&mut self) -> Option<Token<'a>> {
         let start = self.position;
+
         while self.peek_char().is_ascii_digit() {
             self.read_char();
         }
 
+        let p = self.peek_char();
+        match p {
+            '.' => self.read_decimal_part(start),
+            'e' => self.read_exponent(start),
+            'E' => self.read_exponent(start),
```

Comment:
I can change this to match against 'e' | 'E', which will tidy things up a bit.

Reply:
SGTM